operator/cmd/syncclusterconfig/sync.go

```diff
@@ -125,3 +135,83 @@ func loadBoostrapYAML(path string) (map[string]any, error) {
 
 	return config, nil
 }
+
+// maybeMergeSuperusers merges the values found in users.txt into our superusers
+// list found in our bootstrap.yaml. This only occurs if an entry for superusers
+// actually exists in the bootstrap.yaml.
+func maybeMergeSuperusers(logger logr.Logger, clusterConfig map[string]any, path string) {
+	if path == "" {
+		// we have no path to a users.txt, so don't do anything
+		logger.Info("--users-txt not specified. Skipping superusers merge.")
+		return
+	}
+
+	superusersConfig, ok := clusterConfig[superusersEntry]
+	if !ok {
+		// we have no superusers configuration, so don't do anything
+		logger.Info("Configuration does not contain a 'superusers' entry. Skipping superusers merge.")
+		return
+	}
+
+	superusers, err := loadUsersTxt(path)
+	if err != nil {
+		logger.Info(fmt.Sprintf("Error reading users.txt file %q: %v. Skipping superusers merge.", path, err))
+		return
+	}
+
+	superusersAny, ok := superusersConfig.([]any)
+	if !ok {
+		logger.Info(fmt.Sprintf("Unable to cast superusers entry to array. Skipping superusers merge. Type is: %T", superusersConfig))
+		return
+	}
+
+	clusterConfig[superusersEntry] = normalizeSuperusers(append(superusers, mapConvertibleTo[string](superusersAny)...))
+}
+
+func loadUsersTxt(path string) ([]string, error) {
+	usersTxt, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	users := []string{}
+	scanner := bufio.NewScanner(bytes.NewReader(usersTxt))
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		tokens := strings.Split(line, ":")
+		if len(tokens) != 3 {
```

Comment:
Looks like tokens can be of len 2 or 3 (mechanism can be omitted): https://github.com/redpanda-data/helm-charts/blob/56e8cc7a5e8a34ed1a27c8073490e76493ecda86/charts/redpanda/secrets.go#L308-L317